### PR TITLE
interp: weaken panics to errors and return panicked values

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

Updates #413.
Updates #487.

### CODE REVIEW

File: interp/interp.go [hunk 1]
```diff
@@ -289,10 +305,15 @@ func (interp *Interpreter) main() *node {
 	return nil
 }
 
-// Eval evaluates Go code represented as a string. It returns a map on
-// current interpreted package exported symbols
-func (interp *Interpreter) Eval(src string) (reflect.Value, error) {
-	var res reflect.Value
+func (interp *Interpreter) Eval(src string) (res reflect.Value, err error) {
```
Comment (james-lawrence) [inline interp/interp.go]: you lost the method documentation.

Comment (kortschak) [reply interp/interp.go]: Eep. Fixed.

**Comment by mvertes** [comment]
Thank you for your efforts. Sorry to bring my response a bit late in the process.

I have to say that  I'm not convinced that recovering panics _inside_ `Eval` is a better practice than recovering it _outside_, i.e. in the host application (if necessary). My rationale is:
- as panic can be recovered outside of `Eval` anyway, it is not to me a showstopper if it is not implemented in yaegi, the user can always achieve its desired behavior at a small cost.
- panics may be of different origins, system problem, bug in yaegi,, script action. etc. I'm not sure that transforming all these in a yaegi error is the right thing to do. In fact I don't know the best thing to do (and can't), and therefore the best (to me) is still to let the panic propagate, and let the user decide based on context.
- The context in panic is a full stack trace, invaluable for debugging and post-mortem (if no recover), and much richer than a custom error. With the current patch, I loose the ability to debug, and forbid the user to debug as well. Or we could generate stack trace part of error info, but in this case, why not just propagate panics ?

**Comment by mvertes** [comment]
I just went through the PR again, and noticed that you indeed capture the info necessary to access the stack traces, sorry for having missed that. Although it doesn't change the conclusion.

**Comment by kortschak** [comment]
I differ in opinion. What you find is that basically all uses of `Eval` without this then must be wrapped in a recover in order to avoid bringing down the application.

**Comment by kortschak** [comment]
Note that all the things that a user can do with recovery they user can do with a type assertion on the returned `error` from `Eval` and then introspection on the `interp.Panic.Value`. With some additional work, the user could even have a stack trace of the interpreted code (which would not be possible at all with the propagated panic).

**Comment by kortschak** [comment]
*ping*

**Comment by mvertes** [comment]
I added a formatted stack buffer (as returned by `runtime/debug.Stack()`) to preserve the previous tracing behaviour, and still be able to recover automatically. `Panic.Callers` could indeed be used to dump a call trace at interpreter level in the future.

**Comment by kortschak** [comment]
Thanks

**Pull request merged**